src/coreclr/src/vm/multicorejitplayer.cpp

```diff
@@ -624,37 +624,47 @@ void MulticoreJitProfilePlayer::JITMethod(Module * pModule, unsigned methodIndex
     // Similar to Module::FindMethod + Module::FindMethodThrowing,
     // except it calls GetMethodDescFromMemberDefOrRefOrSpec with strictMetadataChecks=FALSE to allow generic instantiation
     MethodDesc * pMethod = MemberLoader::GetMethodDescFromMemberDefOrRefOrSpec(pModule, token, NULL, FALSE, FALSE);
-
-    if ((pMethod != NULL) && ! pMethod->IsDynamicMethod() && pMethod->HasILHeader())
+    if (pMethod != NULL && !pMethod->IsDynamicMethod())
     {
-        // MethodDesc::FindOrCreateTypicalSharedInstantiation is expensive, avoid calling it unless the method or class has generic arguments
-        if (pMethod->HasClassOrMethodInstantiation())
+        if (pMethod->HasILHeader())
         {
-             pMethod = pMethod->FindOrCreateTypicalSharedInstantiation();
+            // MethodDesc::FindOrCreateTypicalSharedInstantiation is expensive, avoid calling it unless the method or class has generic arguments
+            if (pMethod->HasClassOrMethodInstantiation())
+            {
+                pMethod = pMethod->FindOrCreateTypicalSharedInstantiation();
+
+                if (pMethod == NULL)
+                {
+                    goto BadMethod;
+                }
+
+                pModule = pMethod->GetModule_NoLogging();
+            }
 
-            if (pMethod == NULL)
+            if (pMethod->GetNativeCode() != NULL) // last check before
             {
-                goto BadMethod;
+                m_stats.m_nHasNativeCode ++;
+
+                return;
             }
+            else
+            {
+                m_busyWith = methodIndex;
 
-            pModule = pMethod->GetModule_NoLogging();
-        }
+                bool rslt = CompileMethodDesc(pModule, pMethod);
 
-        if (pMethod->GetNativeCode() != NULL) // last check before
-        {
-            m_stats.m_nHasNativeCode ++;
+                m_busyWith = EmptyToken;
 
-            return;
+                if (rslt)
+                {
+                    return;
+                }
+            }
         }
         else
         {
-            m_busyWith = methodIndex;
-
-            bool rslt = CompileMethodDesc(pModule, pMethod);
-
-            m_busyWith = EmptyToken;
-
-            if (rslt)
+            // NDirect Stub
```

Comment:
The `else` clause above should be `else if (pMethod->IsNDirect())` instead

Reply:
Fixed. Thank you.

Reply:
Hmm I'm not seeing the fix, I see the commit that fixed it but it didn't seem to have made it into the branch somehow

Reply:
Oops. I am sorry. I made a mistake. Thank you.